client/src/main/java/com/kritsit/casetracker/client/domain/factory/ServiceFactory.java

```diff
@@ -15,7 +15,7 @@
     private static final Logger logger = LoggerFactory.getLogger(ServiceFactory.class);
     private static IConnectionService connection;
 
-    public static IConnectionService getServerConnection() {
+    public static synchronized IConnectionService getServerConnection() {
```

Comment:
"Incorrect lazy initialization of static field." It is worried about a thread calling it while another thread is still processing the instantiation. Adding synchronized fixes this.